processing/src/main/java/org/apache/druid/segment/nested/NestedCommonFormatColumn.java

```diff
@@ -75,9 +75,11 @@ default Indexed<Object[]> getArrayDictionary()
 
   default SortedMap<String, FieldTypeInfo.MutableTypeSet> getFieldTypeInfo()
   {
-    FieldTypeInfo.MutableTypeSet rootOnlyType = new FieldTypeInfo.MutableTypeSet().add(getLogicalType());
     SortedMap<String, FieldTypeInfo.MutableTypeSet> fields = new TreeMap<>();
-    fields.put(NestedPathFinder.JSON_PATH_ROOT, rootOnlyType);
+    if (!getLogicalType().equals(ColumnType.NESTED_DATA)) {
```

Comment:
nit: invert it so that there's 0 chance of an NPE.  I.e. `ColumnType.NESTED_DATA.equals(getLogicalType())` cannot NPE on this line